src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -28,6 +46,7 @@ public class TagHelperDescriptor
         /// Instantiates a new instance of the <see cref="TagHelperDescriptor"/> class with the given
         /// <paramref name="attributes"/>.
         /// </summary>
+        /// <param name="prefix">The prefix of the HTML element name the tag helper targets.</param>
```

Comment:
First bullet point: `TagHelperDescriptor`'s building process logic exists in the `TagHelperDescriptorResolver`; therefore I feel that anything that affects them should also exist in the `TagHelperDescriptorResolver`, not to mention they all hold the same basic string format/directive structure. 

For your second bullet point I don't see the value in abstracting the `Prefix` into a combination of the element name + prefix from a `TagHelperDescriptor`s perspective. The `TagHelperDescriptor` should be extremely dumb/purely informational, consumers should deal with understanding what it applies to in the document.